contracts/sysio.roa/include/sysio.roa.hpp

```diff
@@ -225,6 +250,34 @@ namespace sysio {
             >nodeownerreg_t;
 
 
+            /**
+             * @brief Table mapping (scoped by creator) of nonce to the created username.
+             */
+            struct [[sysio::table]] sponsor {
+                name nonce;
+                name username;
+
+                uint64_t primary_key() const { return nonce.value; }
+                uint64_t by_username() const { return username.value; }
+            };
+
+            typedef sysio::multi_index<
+                "sponsors"_n, sponsor,
+                indexed_by<"byusername"_n, const_mem_fun<sponsor, uint64_t, &sponsor::by_username>>
+            > sponsors_t;
+
+            /**
+             * @brief Table tracking how many new users a node owner has sponsored.
+             */
+            struct [[sysio::table]] sponsorcount {
+                name owner;
+                uint64_t count;
+
+                uint64_t primary_key() const { return owner.value; }
+            };
+
+            typedef sysio::multi_index<"sponsorcount"_n, sponsorcount> sponsorcount_t;
```

Comment:
I assume this is for external view of how many users a sponsor has sponsored. This could be done via a read-only transaction that just returns count of `by_username` of the `sponsor` table.